src/models/wasteWater/story/WasteWaterSamplingSites.tsx

```diff
@@ -17,7 +18,7 @@ export const WasteWaterSamplingSites = () => {
     return <Loader />;
   }
 
-  const dataInTimeRange = filterByDateRange(wasteWaterData, dateRangeSelector.getDateRange());
+  const dataInTimeRange = filterByDateRange(wasteWaterData, dateRangeSelector.getDateRange()).filter((ds) => !legacyLocations.has(ds.location));
```

Comment:
Instead of duplicating the function, this should be a wrapper into a common function and the filter should be passed as a lamba.